### PR TITLE
Support V2 and V3 calling simultanously. Configurable in developer settings

### PR DESCRIPTION
**Pull request by typfel**
This PR introduces support for calling protocol 3. This support lives alongside the old calling protocol and in the developer setting you can configure which protocol you want use for outgoing calls.

Needs release from:
- [x]  https://github.com/wireapp/wire-ios-sync-engine/pull/142
- [x]  https://github.com/wireapp/wire-ios-data-model/pull/89

### CODE REVIEW

File: Wire-iOS/Sources/Helpers/syncengine/ZMConversation+Additions.m [hunk 1]
```diff
@@ -329,10 +329,12 @@ - (void)joinVoiceChannelWithoutAskingForPermissionWithVideo:(BOOL)video completi
             __block BOOL joined = YES;
             [[ZMUserSession sharedSession] enqueueChanges:^{
                 if (video) {
-                    joined = [self.voiceChannel joinVideoCall:nil inUserSession:[ZMUserSession sharedSession]];
+//                    joined = [self.voiceChannel joinVideoCall:nil inUserSession:[ZMUserSession sharedSession]];
```
Comment (mikeger) [inline Wire-iOS/Sources/Helpers/syncengine/ZMConversation+Additions.m]: // FIXME

File: Wire-iOS/Sources/Managers/SoundEventListener.m [hunk 2]
```diff
@@ -258,15 +258,15 @@ - (void)changeObservedVoiceChannelConversation:(ZMVoiceChannel *)voiceChannel
     } else {
         if (voiceChannel.state == ZMVoiceChannelStateSelfConnectedToActiveChannel) {
             [self unregisterAsCallParticipantObserver];
-            self.callParticipantsToken = [voiceChannel addCallParticipantsObserver:self];
+//            self.callParticipantsToken = [ZMVoiceChannel addCallParticipantsObserver:self inConversation:voiceChannel.conversation voiceChannel:voiceChannel]; // FIXME
```
Comment (mikeger) [inline Wire-iOS/Sources/Managers/SoundEventListener.m]: // FIXME

Comment (typfel) [inline Wire-iOS/Sources/Managers/SoundEventListener.m]: This one has a fix me at the end.

**Comment by marcoconti83** [comment]
Sync engine has been released (some days ago), if that was the last blocker. But why so many commits in this PR? It looks fishy!

**Comment by typfel** [comment]
I will close this PR. This branch is not used anymore. I'm working on `feature/calling-v3-base`

**Comment by typfel** [comment]
It's because develop has been merged into `feature/calling-v3-dualstack`. I'll rebase the onto develop like I did with the SE calling branches.

File: ConfigurationSets/Base.xcconfig [hunk 3]
```diff
@@ -43,7 +43,8 @@ DEBUG_INFORMATION_FORMAT = dwarf-with-dsym
 
 // Search Paths
 //
-AVS_ROOT = $(SRCROOT)/Libraries/wire-avs-ios
+//AVS_ROOT = $(SRCROOT)/Libraries/wire-avs-ios
```
Comment (mikeger) [inline ConfigurationSets/Base.xcconfig]: I guess the commented part is not needed

Comment (typfel) [inline ConfigurationSets/Base.xcconfig]: Ah, shouldn't be needed at all now I think.

Comment (mikeger) [inline ConfigurationSets/Base.xcconfig]: Also I guess you can remove "$(AVS_ROOT)/include" and $(AVS_ROOT)/lib from the search paths

**Comment by mikeger** [comment]
@typfel FYI has a conflict with Cartfile and project file

**Pull request merged**